interfaces/builtin/serial_port.go

```diff
@@ -50,25 +62,59 @@ func (iface *SerialPortInterface) SanitizeSlot(slot *interfaces.Slot) error {
 		panic(fmt.Sprintf("slot is not of interface %q", iface))
 	}
 
+	// We will only allow creation of this type of slot by a gadget or OS snap
+	if !(slot.Snap.Type == "gadget" || slot.Snap.Type == "os") {
+		return fmt.Errorf("serial-port slots only allowed on gadget or core snaps")
+	}
+
 	// Check slot has a path attribute identify serial device
 	path, ok := slot.Attrs["path"].(string)
 	if !ok || path == "" {
 		return fmt.Errorf("serial-port slot must have a path attribute")
 	}
 
-	// Clean the path before checking it matches the pattern
+	// Clean the path before further checks
 	path = filepath.Clean(path)
 
-	// Check the path attribute is in the allowable pattern
-	if serialAllowedPathPattern.MatchString(path) {
-		return nil
+	switch len(slot.Attrs) {
+	case 1:
```

Comment:
As I mentioned both in the prior review and during our meeting, it's not okay to do this. Imagine the next version of this interface accepts one more parameter while preserving compatibility. This would break any chances of it working.

Instead, verify what you need: if there is either a usb-vendor or a usb-product or both, we should ensure both are present (or error) and create the udev rule. The path must always be checked regardless if it is present, otherwise we could put a symlink at an arbitrary place in the system via this interface.